source/fab/language/__init__.py

```diff
@@ -54,6 +54,30 @@ def products(self) -> List[Path]:
         return []
 
 
+class HashCalculator(Task):
+    def __init__(self, hasher: TextReaderAdler32, database: StateDatabase):
+        self._hasher = hasher
+        self._database = database
+
+    def run(self):
+        for _ in self._hasher.line_by_line():
+            pass  # Make sure we've read the whole file
+        file_info = FileInfoDatabase(self._database)
+        file_info.add_file_info(Path(self._hasher.filename),
+                                self._hasher.hash)
+
+    @property
+    def prerequisites(self) -> List[Path]:
+        if isinstance(self._hasher.filename, Path):
+            return [self._hasher.filename]
+        else:
+            return []
+
+    @property
+    def products(self) -> List[Path]:
+        return []
+
+
```

Comment:
I wonder if we're putting too much int `__init__.py`. It's a magic file, we probably shouldn't be over using it.

Reply:
Yeah you may be write, I suppose we could have a `fab.language.generic`?

Reply:
Following some discussion, `language` has been renamed to `tasks` and we now have `fab.tasks.common` which contains `HashCalculator` and `CommandTask` - the rest has stayed in `__init__.py`